src/Renci.SshNet/Connection/HttpConnector.cs

```diff
@@ -29,18 +29,40 @@ namespace Renci.SshNet.Connection
     ///   </item>
     /// </list>
     /// </remarks>
-    internal sealed class HttpConnector : ProxyConnector
+    internal sealed partial class HttpConnector : ProxyConnector
     {
+        private const string HttpResponsePattern = @"HTTP/(?<version>\d[.]\d) (?<statusCode>\d{3}) (?<reasonPhrase>.+)$";
+        private const string HttpHeaderPattern = @"(?<fieldName>[^\[\]()<>@,;:\""/?={} \t]+):(?<fieldValue>.+)?";
+
+#if NET7_0_OR_GREATER
+        [GeneratedRegex(HttpResponsePattern)]
+        private static partial Regex HttpResponseRegex();
+
+        [GeneratedRegex(HttpHeaderPattern)]
+        private static partial Regex HttpHeaderRegex();
+#else
+        private static readonly Regex HttpResponseRe = new Regex(HttpResponsePattern, RegexOptions.Compiled);
+
+        private static Regex HttpResponseRegex()
+        {
+            return HttpResponseRe;
+        }
+
+        private static readonly Regex HttpHeaderRe = new Regex(HttpHeaderPattern, RegexOptions.Compiled);
+
+        private static Regex HttpHeaderRegex()
+        {
+            return HttpHeaderRe;
+        }
+#endif
```

Comment:
Slightly more compact this way?

```suggestion
#if NET7_0_OR_GREATER
        private static readonly Regex HttpResponseRegex = GetHttpResponseRegex();
        private static readonly Regex HttpHeaderRegex = GetHttpHeaderRegex();

        [GeneratedRegex(HttpResponsePattern)]
        private static partial Regex GetHttpResponseRegex();

        [GeneratedRegex(HttpHeaderPattern)]
        private static partial Regex GetHttpHeaderRegex();
#else
        private static readonly Regex HttpResponseRegex = new Regex(HttpResponsePattern, RegexOptions.Compiled);
        private static readonly Regex HttpHeaderRegex = new Regex(HttpHeaderPattern, RegexOptions.Compiled);
#endif
```

Reply:
Makes sense, pushed-